src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/YFlowUpdateService.java

```diff
@@ -0,0 +1,387 @@
+     *
+     * @param key command identifier.
+     * @param request request data.
+     */
+    public void handlePartialUpdateRequest(String key, CommandContext commandContext, YFlowPartialUpdateRequest request)
+            throws DuplicateKeyException {
+        YFlowRequest target = null;
+        if (request.getYFlowId() != null) {
+            YFlow yFlow = yFlowRepository.findById(request.getYFlowId()).orElse(null);
+            target = YFlowRequestMapper.INSTANCE.toYFlowRequest(yFlow);
+        }
+
+        if (target == null) {
+            target = YFlowRequest.builder().build();
+        }
+
+        Optional.ofNullable(request.getYFlowId()).ifPresent(target::setYFlowId);
+        if (request.getSharedEndpoint() != null) {
+            if (target.getSharedEndpoint() == null) {
+                target.setSharedEndpoint(new FlowEndpoint(request.getSharedEndpoint().getSwitchId(),
+                        request.getSharedEndpoint().getPortNumber()));
+            } else {
+                SwitchId switchId = Optional.ofNullable(request.getSharedEndpoint().getSwitchId())
+                        .orElse(target.getSharedEndpoint().getSwitchId());
+                int portNumber = Optional.ofNullable(request.getSharedEndpoint().getPortNumber())
+                        .orElse(target.getSharedEndpoint().getPortNumber());
+                target.setSharedEndpoint(new FlowEndpoint(switchId, portNumber));
+            }
+        }
+        Optional.ofNullable(request.getMaximumBandwidth()).ifPresent(target::setMaximumBandwidth);
+        Optional.ofNullable(request.getPathComputationStrategy()).ifPresent(target::setPathComputationStrategy);
+        Optional.ofNullable(request.getEncapsulationType()).ifPresent(target::setEncapsulationType);
+        Optional.ofNullable(request.getMaxLatency()).ifPresent(target::setMaxLatency);
+        Optional.ofNullable(request.getMaxLatencyTier2()).ifPresent(target::setMaxLatencyTier2);
+        Optional.ofNullable(request.getIgnoreBandwidth()).ifPresent(target::setIgnoreBandwidth);
+        Optional.ofNullable(request.getPeriodicPings()).ifPresent(target::setPeriodicPings);
+        Optional.ofNullable(request.getPinned()).ifPresent(target::setPinned);
+        Optional.ofNullable(request.getPriority()).ifPresent(target::setPriority);
+        Optional.ofNullable(request.getStrictBandwidth()).ifPresent(target::setStrictBandwidth);
+        Optional.ofNullable(request.getDescription()).ifPresent(target::setDescription);
+        Optional.ofNullable(request.getAllocateProtectedPath()).ifPresent(target::setAllocateProtectedPath);
+
+        if (request.getSubFlows() != null && !request.getSubFlows().isEmpty()) {
+            Map<String, SubFlowDto> stringSubFlowDtoMap = new HashMap<>();
+            if (target.getSubFlows() != null) {
+                stringSubFlowDtoMap = target.getSubFlows().stream()
+                        .collect(Collectors.toMap(SubFlowDto::getFlowId, Function.identity()));
```

Comment:
Produce exception on provided sub flow's flow_id collision. Sould we return "reasonable" error in this case instead of just "INTERNAL_ERROR"?

Reply:
Exceptions have been added.